src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -1806,9 +1806,11 @@ private Field DetermineMinimalOptionalField(Node nd)
             }
             else
             {
-                // otherwise, if there is a single optional node, use that..
+                // otherwise, if there are optional nodes, use that..
                 int nodeCount = nd.Fields.Count(f => IsNode(f.Type) && f.Type != "SyntaxToken");
-                if (nodeCount == 1)
+                // AccessorDeclarationSyntax is the only node hat has nodeCount == 2 and must be handled
+                // as well to return a correct method signature
+                if (nodeCount == 1 || nodeCount == 2)
```

Comment:
I think it would be preferred to preserve that API by explicit move to a partial declaration rather than making changes to the syntax generator.